src/command.rs

```diff
@@ -49,6 +51,95 @@ macro_rules! delegate {
     }};
 }
 
+/// If a command is `OverSsh` then it can be executed over an SSH session.
+/// Primarily a way to allow `std::process::Command` to be turned directly into an `openssh::Command`.
+pub trait OverSsh {
+    /// Given a session, return a command that can be executed over that session.
+    ///
+    /// **Note**: The command to be executed on the remote machine does not include
+    /// any environment variables or the current directory. They must be
+    /// set explicitly, if desired.
+    ///
+    /// # Example
+    /// Consider the implementation of `OverSsh` for `std::process::Command`:
+    ///
+    /// ```no_run
+    /// # #[tokio::main(flavor = "current_thread")]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     use std::process::Command;
+    ///     use openssh::{Session, KnownHosts, OverSsh};
+    ///
+    ///     let session = Session::connect_mux("me@ssh.example.com", KnownHosts::Strict).await?;
+    ///     let ls =
+    ///         Command::new("ls")
+    ///         .arg("-l")
+    ///         .arg("-a")
+    ///         .arg("-h")
+    ///         .over_session(&session)
+    ///         .output()
+    ///         .await?;
+    ///
+    ///     assert!(String::from_utf8(ls.stdout).unwrap().contains("total"));
+    /// #   Ok(())
+    /// }
+    ///
+    /// ```
+    fn over_session<'session>(&self, session: &'session Session) -> crate::Command<'session>;
```

Comment:
It's generally common in Rust to have the method of a single-purpose trait match the name of that trait. See for example `Clone::clone`, `AsRef::as_ref`, `Hash::hash`, etc. So we should either name this trait `OverSession` or name the method `over_ssh`. I _think_ I prefer the latter, because it's clearer what's going on from reading just the code. We _could_ also make it `OverSsh::over_ssh_session`, though the verbosity is a little unfortunate.

Reply:
I do believe `OverSsh::over_ssh` makes it quite a bit clearer. That's an excellent improvement. Thank you.